x/evm/keeper/msg_server.go

```diff
@@ -323,3 +324,22 @@ func (server msgServer) RegisterPointer(goCtx context.Context, msg *types.MsgReg
 	}
 	return &types.MsgRegisterPointerResponse{PointerAddress: pointerAddr.String()}, err
 }
+
+func (server msgServer) AssociateContractAddress(goCtx context.Context, msg *types.MsgAssociateContractAddress) (*types.MsgAssociateContractAddressResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	addr := sdk.MustAccAddressFromBech32(msg.Address) // already validated
+	// check if address is for a contract
+	if server.wasmViewKeeper.GetContractInfo(ctx, addr) == nil {
+		return nil, errors.New("no wasm contract found at the given address")
+	}
+	evmAddr := common.BytesToAddress(addr)
+	existingEvmAddr, ok := server.GetEVMAddress(ctx, addr)
+	if ok {
```

Comment:
do we need to handle the case where ok == false ?

Reply:
that's the case where we set the association (because no existing association)